### PR TITLE
Add stem leaf plot example

### PR DESCRIPTION
**Pull request by eitanlees**
My attempt at a stem and leaf plot.

### CODE REVIEW

**Comment by jakevdp** [comment]
I think you could do the grouping with a groupby, not sure if this is more clear or more obfuscated :smile:

```python
grouped_data = original_data.groupby('stem')['leaf'].apply(lambda leaves: ''.join(sorted(map(str, leaves))))
```

I wonder if there would be a better way to do this by actually encoding the x positions of the digits?

**Comment by eitanlees** [comment]
haha, nice one liner! 

Obfuscation always creeps in the longer I work on a problem. It's the [curse of knowledge](https://en.wikipedia.org/wiki/Curse_of_knowledge)! 

With these examples it's sometimes tough to strike the balance between "Hey look at this cool thing I did" vs "This is a building block of visualization"

I will go back to the drawing board and think of how to encode the positions.

**Comment by eitanlees** [comment]
I think calculating positions is better than just plotting a concatenated string. This is a first pass at trying to calculate the positions of the leaves.

**Comment by jakevdp** [comment]
This is nice! Though I generally think it's clearer to use ``apply`` rather than looping over groups; something like this:

```python
get_position = lambda x: 1 + pd.Series(range(len(x)))
    
original_data['position'] = = original_data.groupby('stem')\
                                           .apply(get_position)\
                                           .reset_index(level=1, drop=True)
```

**Comment by eitanlees** [comment]
I see, I haven't gotten comfortable with `apply` yet and often resort to loops.

File: altair/vegalite/v2/examples/stem_and_leaf.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# Determining leaf position
+get_position = lambda x: 1 + pd.Series(range(len(x)))
+    
+original_data['position'] = original_data.groupby('stem')\
+                                         .apply(get_position)\
+                                         .reset_index(drop=True)
+
+# Creating stem and leaf plot
+chart = alt.Chart(original_data).mark_text(align='left', baseline='middle', dx=-5).encode(
```
Comment (jakevdp) [inline altair/vegalite/v2/examples/stem_and_leaf.py]: Formatting-wise, I'd break up the ``mark_text`` args:
```python
chart = alt.Chart(original_data).mark_text(
    align='left',
    baseline='middle',
    dx=-5
).encode(
    ...
```

**Comment by jakevdp** [comment]
Looks pretty good!

**Comment by jakevdp** [comment]
Looks good to me. If you're happy with it, I'll merge!

**Comment by eitanlees** [comment]
👍

**Pull request merged**